File: docs-examples/example-java/src/test/java/io/micronaut/rabbitmq/docs/consumer/executor/CustomExecutorSpec.java
package io.micronaut.rabbitmq.docs.consumer.executor;

import io.micronaut.context.annotation.Property;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import io.micronaut.test.support.TestPropertyProvider;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;

import java.util.HashMap;
import java.util.Map;

import static java.util.concurrent.TimeUnit.SECONDS;
import static org.awaitility.Awaitility.await;
import io.micronaut.rabbitmq.testcontainers.RabbitMQ;

@MicronautTest(rebuildContext = true)
@Property(name = "spec.name", value = "CustomExecutorSpec")
@Property(name = "micronaut.executors.product-listener.type", value = "FIXED")
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class CustomExecutorSpec implements TestPropertyProvider {
    @Override
    public Map<String, String> getProperties() {
        return RabbitMQ.getProperties();
    }

    @Test
    void testProductClientAndListener(ProductClient productClient, ProductListener productListener) {

// tag::producer[]
productClient.send("custom-executor-test".getBytes());
// end::producer[]

        await().atMost(60, SECONDS).until(() ->
                productListener.messageLengths.size() == 1 &&
                productListener.messageLengths.get(0).equals("custom-executor-test")
        );
    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/rabbitmq/docs/consumer/executor/CustomExecutorSpec.kt
package io.micronaut.rabbitmq.docs.consumer.executor

import io.kotest.assertions.timing.eventually
import io.kotest.core.spec.style.AnnotationSpec
import io.kotest.matchers.shouldBe
import io.micronaut.context.annotation.Property
import io.micronaut.rabbitmq.testcontainers.RabbitMQ
import io.micronaut.test.extensions.kotest5.annotation.MicronautTest
import io.micronaut.test.support.TestPropertyProvider
import jakarta.inject.Inject
import kotlin.time.Duration.Companion.seconds

@MicronautTest(rebuildContext = true)
@Property(name = "micronaut.executors.product-listener.type", value = "FIXED")
@Property(name = "spec.name", value = "CustomExecutorSpec")
class CustomExecutorSpec : TestPropertyProvider, AnnotationSpec() {
    override fun getProperties(): Map<String, String> {
        return RabbitMQ.getProperties()
    }

    @Inject
    lateinit var productClient: ProductClient

    @Inject
    lateinit var productListener: ProductListener

    @Test
    suspend fun testBasicConsumerAndProducer() {


// tag::producer[]
        productClient.send("custom-executor-test".toByteArray())
// end::producer[]

        eventually(10.seconds) {
            productListener.messageLengths.size shouldBe 1
            productListener.messageLengths[0] shouldBe "custom-executor-test"
        }
    }
}
